Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/ADLSGen2Deploy.java

```diff
@@ -93,13 +92,7 @@ private URI getUploadDir() {
                 .flatMap(ignore -> op.createFile(filePath))
                 .flatMap(ignore -> op.uploadData(filePath, src))
                 .doOnNext(ignore -> log().info(String.format("Append data to file %s successfully.", filePath)))
-                .map(ignored -> {
-                    try {
-                        return getArtifactUploadedPath(filePath);
-                    } catch (URISyntaxException ex) {
-                        throw new RuntimeException(new IllegalArgumentException("Can not get valid artifact upload path" + ex.toString()));
-                    }
-                });
+                .map(ignored -> AbfsUri.parse(filePath).getUri().toString());
```

Comment:
To align with method `getArtifactUploadedPath`, we should use `AbfsUri.parse(filePath).getPath()`. Forget it if this change is by design.